src/screens/IndexScreen.js

```diff
@@ -120,9 +133,14 @@ export const IndexScreen = ({ navigation, route }) => {
       [QUERY_TYPES.NEWS_ITEMS]: showNewsFilter
     }[query];
 
-  const hasCategoryFilterSelection = !!Object.prototype.hasOwnProperty.call(queryVariables, [
-    keyForSelectedValueByQuery?.[query]
-  ]);
+  const hasCategoryFilterSelection = useCallback(
+    (query, isLocationFilter) => {
```

Comment:
how is the location filter related to the category filter?
i guess it needs an own selection callback keeping the `hasCategoryFilterSelection` untouched.

Reply:
The problem here is that the `query` does not change. The `query` remains the same but the `queryVariables` change. Since the query remains the same, it is difficult to understand whether it is a `category` filter or a `location` filter.

Reply:
now we have an extra query for addresses and a separate branch has been created for the refactor as we discussed.
https://github.com/smart-village-solutions/smart-village-app-app/commit/88d315d19937921c6729c08bef7e3d3ba2e03260